.github/workflows/build.yml

```diff
@@ -318,3 +321,38 @@ jobs:
       run: make pythoninfo
     - name: Tests
       run: xvfb-run make buildbottest TESTOPTS="-j4 -uall,-cpu"
+
+  check:  # This job does nothing and is only used for the branch protection
+    if: always()
+
+    needs:
+    - check_source  # Transitive dependency, needed to access `run_tests` value
+    - check_generated_files
+    - build_win32
+    - build_win_amd64
+    - build_macos
+    - build_ubuntu
+    - build_ubuntu_ssltests
+    - build_asan
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@13b4244b312e8a314951e03958a2f91519a6a3c9
```

Comment:
> What's the reason this is pinned to v1.2.1 instead of using the release name?

No particular reason from my side but when I contributed this to other projects, some wanted to have commit sha pinning. Dependabot is able to bump such commit hash based pins so it's usually considered the best of both worlds.

For myself, I usually do this:
```suggestion
      uses: re-actors/alls-green@release/v1
```
But I appreciate that some people want better traceability for non-GH-official actions. It's up to you really.

If you choose to go for the `release/v1` branch, it represents stable API. Note that Dependabot will not be updating the reference in this case.

> Also, I looked at the repo of the action and the checks are red there, is that expected?

Yes, kinda. It is necessary for internal smoke-testing. Some of the jobs are red but the check job isn't. The overall workflow status may be red but it's not necessary a reflection of the gate check, it's just a quirk of how GitHub represents statuses — it doesn't know that we actually care about the outcome of one single job which decides internally whether the dependencies are allowed to fail or be skipped, and if any of the workflow jobs is red, it marks the whole thing as red. The branch protection is a disconnected concept and is not applied in all possible contexts, which is why GH can't make assumptions automatically, and it doesn't provide a way to declare what's important on the GHA declaration level.
It's weird but we can't do anything about it, unless GitHub decides to make updates...